src/faiss.cc

```diff
@@ -335,11 +472,149 @@ class IndexFlatL2 : public Napi::ObjectWrap<IndexFlatL2>
     delete[] xb;
     return Napi::Number::New(info.Env(), num);
   }
+
+  Napi::Value toBuffer(const Napi::CallbackInfo &info)
+  {
+    Napi::Env env = info.Env();
+
+    if (info.Length() != 0)
+    {
+      Napi::Error::New(env, "Expected 0 arguments, but got " + std::to_string(info.Length()) + ".")
+          .ThrowAsJavaScriptException();
+      return env.Undefined();
+    }
+
+    faiss::VectorIOWriter *writer = new faiss::VectorIOWriter();
+
+    faiss::write_index(index_.get(), writer);
+
+    // return buffer from IOWriter
+    return Napi::Buffer<uint8_t>::Copy(env, writer->data.data(), writer->data.size());
+  }
+
+  static Napi::Object NewInstance(Napi::Env env, const std::vector<napi_value> &args)
+  {
+    Napi::EscapableHandleScope scope(env);
+    Napi::Object obj = GetInstanceData(env, T::CLASS_NAME).New(args);
```

Comment:
I found before in this experimental branch that it is simpler to store the pointer to the constructor using the following way:
https://github.com/ewfian/faiss-node/blob/feat/support-IndexFlatIP/src/faiss.cc#L71

`inline static Napi::FunctionReference *constructor = new Napi::FunctionReference();`

This eliminates the need to use `GetInstanceData` and `AttachInstanceData`

Please confirm if you can modify it as above?

Reply:
My original changes had used that pattern but was having stability issues with the Bun runtime so I opted for the more complex pattern. I later learned the Bun issues were due to a bug anyway, so now its reverted to use the simpler pattern.